### PR TITLE
Analysis trade area

### PR DESCRIPTION
**Pull request by dgaubert**
cc @rochoa

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
```
Comment (rochoa) [inline package.json]: Do not bump version neither generate reference in PR.

**Comment by rochoa** [comment]
We are missing the `disolved` option.

File: lib/node/nodes/trade-area/trade-area-query-builder.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+'use strict';
```
Comment (rochoa) [inline lib/node/nodes/trade-area/trade-area-query-builder.js]: I was thinking more about a generic method to load any node/*.sql file as a `dot.template`.

File: lib/node/nodes/trade-area/trade-area.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+'use strict';
```
Comment (rochoa) [inline lib/node/nodes/trade-area/trade-area.js]: I prefer to keep the nodes inside `lib/node/nodes` and not create new directories.

File: lib/node/nodes/trade-area/trade-area.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+var Node = require('../../node');
+var buildQuery = require('./trade-area-query-builder');
+var buildRange = require('./data-range-builder');
```
Comment (rochoa) [inline lib/node/nodes/trade-area/trade-area.js]: I would prefer to keep it everything in here: no builders, etc.

If a node is too complex probably we have to consider splitting it into different nodes.

File: lib/node/node.js [hunk 5]
```diff
@@ -212,6 +214,14 @@ module.exports.getChildNodesNames = function(NodeClass) {
     }, []);
 };
 
+module.exports.getTemplate = function (name) {
+    var template = dot.template(fs.readFileSync(__dirname + '/nodes/sql/' + name +'.sql'));
+
+    return function (it) {
```
Comment (rochoa) [inline lib/node/node.js]: This is the same as `return dot.template(...)`, isn't it?

**Comment by rochoa** [comment]
GO, GO, GO!

**Comment by dgaubert** [comment]
I've just applied changes from `nodes-refactor-tplfn`

File: lib/node/node.js (hunk 5 above)
Comment (dgaubert) [inline lib/node/node.js]: Ups, I don't know what I was thinking.

**Pull request merged**